svg.panzoom.js

```diff
@@ -129,16 +127,49 @@ SVG.extend(SVG.Doc, SVG.Nested, {
       lastP = currentP
     }
 
-    this.on('wheel', zoom)
+    this.on('wheel', wheelZoom)
     this.on('touchstart', pinchZoomStart, this, {passive:false})
     this.on('mousedown', panStart, this, {passive:false})
     this.on('touchstart', panStart, this, {passive:false})
 
     return this
 
-  }
+  },
+
+  zoom: function(level, point) {
+    var style = window.getComputedStyle(this.node)
+      , width = parseFloat(style.getPropertyValue('width'))
+      , height = parseFloat(style.getPropertyValue('height'))
+      , v = this.viewbox()
+      , zoomX = width / v.width
+      , zoomY = height / v.height
+      , zoom = Math.min(zoomX, zoomY)
+
+    if(level == null) {
+      return zoom
+    }
 
+    var zoomAmount = (zoom / level)
+
+    point = point || new SVG.Point(width/2 / zoomX + v.x, height/2 / zoomY + v.y)
+
+    var box = new SVG.Box(v)
+      .transform(new SVG.Matrix()
+        .scale(zoomAmount, point.x, point.y)
+      )
+
+    if(this.fire('zoom', {level: level, box: box, focus: point}).event().defaultPrevented)
```

Comment:
In svg.js we lately introduced the `event()` method which returns the last fired event of that element. By firing an event and immediately checking the getting that event, we could check the `defaultPrevented` flag. It is true, if any event handler called `preventDefault()` on the event.

So in case it *is* true, we return because our behavior should not get executed